### PR TITLE
Add basic CI job that runs CMake builds

### PR DESCRIPTION
**Pull request by chiphogg**
For now, we simply run "the default CMake generator on the latest
Ubuntu".  Later on, if we tighten up our notion of "supported CMake
configurations", we can add more options here.

Helps #215.

### CODE REVIEW

**Comment by chiphogg** [comment]
FYI @timhirsh, we'll need a security review on jwlawson/actions-setup-cmake@802fa1a2c4e212495c05bf94dba2704a92a472be.  Assuming it passes, we can add it to the allowlist in the repo settings, and after that, the job should run remotely.  (I already ran it locally using [nektos/act](https://github.com/nektos/act), so I know it works.)

**Comment by chiphogg** [comment]
Alternatively, perhaps we should consider batching the security reviews (see #260).  Our version of `checkout`, for example, is pretty old.

File: .github/workflows/cmake-build-and-test.yml
Comment (timhirsh) [inline .github/workflows/cmake-build-and-test.yml:28]: See #261 for context:
```suggestion
      - uses: actions/checkout@44c2b7a8a4ea60a981eaca3cf939b5f4305c123b #v4.1.5
```

**Review by timhirsh — APPROVED** [review]
This LGTM, pending completion of the security review 👍

**Pull request merged**
